src/modules/login_user/app/login_user_usecase.ts

```diff
@@ -0,0 +1,13 @@
+import { IUserRepository } from '../../../shared/domain/repositories/user_repository_interface'
+import { EntityError } from '../../../shared/helpers/errors/domain_errors'
+
+export class LoginUserUsecase {
+  constructor(private repo: IUserRepository) {}
+
+  async execute(email: string, password: string) {
+    if (!email || !password) {
```

Comment:
It should validate de email and password beyond if its only null or not, i believe that should have methods in the user class to validate them, and also should raise a NoItemsFound if the repository returns Null